src/api/SetTokenAPI.ts

```diff
@@ -152,10 +152,71 @@ export class SetTokenAPI {
     return quantity.mod(naturalUnit).eq(ZERO);
   }
 
+  /**
+   * Calculates the units of components of a Set transferred given an issue or redeem quantity.
+   *
+   * @param  setAddress    Address of the Set
+   * @param  quantity      Quantity of Set issued or redeemed
+   * @return               List of units transferred ordered
+   */
+  public async calculateUnitsTransferred(setAddress: Address, quantity: BigNumber): Promise<BigNumber[]> {
+    this.assertCalculateUnitsTransferred(setAddress, quantity);
+
+    const [naturalUnit, componentUnits] = await Promise.all([
+      this.setToken.naturalUnit(setAddress),
+      this.setToken.getUnits(setAddress),
+    ]);
+
+    return _.map(componentUnits, componentUnit => calculatePartialAmount(componentUnit, quantity, naturalUnit));
+  }
+
+  /**
+   * Calculate the units of a specified component of a Set transferred given an issue or redeem quantity.
+   *
+   * @param  setAddress        Address of the Set
+   * @param  componentAddress  Address of the component
```

Comment:
min 3 spaces between the longest parameter name and the start of the description